env_docker_ext_test.go

```diff
@@ -22,15 +22,15 @@ func wgetFlagsCmd(hostPort string) e2e.Command {
 func TestDockerEnvironment(t *testing.T) {
 	t.Parallel()
 
-	e, err := e2e.NewDockerEnvironment("e2e_lifecycle")
+	e, err := e2e.New()
 	testutil.Ok(t, err)
 	t.Cleanup(e.Close)
 
 	p1 := e2edb.NewPrometheus(e, "prometheus-1")
 	testutil.Equals(t, "prometheus-1", p1.Name())
 	testutil.Equals(t, filepath.Join(e.SharedDir(), "data", p1.Name()), p1.Dir())
 	testutil.Equals(t, filepath.Join("/shared", "data", p1.Name()), p1.InternalDir())
-	testutil.Equals(t, "e2e_lifecycle-prometheus-1:9090", p1.InternalEndpoint("http"))
+	testutil.Equals(t, "e2e-80D5E99C3992-prometheus-1:9090", p1.InternalEndpoint("http"))
```

Comment:
This prefix will be random, no? Might be better to do something like: `fmt.Sprintf("%s-prometheus-1:9090", e.name)`.

Reply:
Ah, it won't be random. My bad. It changes depending on the caller it seems.

Reply:
What if a caller needs two envs though? Would the recommendation be to use the `WithName` option? 

Shouldn't `WithName` run some validations to ensure the names won't break hostname/DNS issues? This can save a lot of headacha.

Reply:
> Would the recommendation be to use the WithName option?

Yes, we can validate indeed.